assignments/prototypes.js

```diff
@@ -32,7 +55,53 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
- 
+
+function Humanoid(humanoidAttrs) {
+  CharacterStats.call(this, humanoidAttrs);
+  this.team = humanoidAttrs.team;
+  this.weapons = humanoidAttrs.weapons;
+  this.language = humanoidAttrs.language;
+};
+
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+Humanoid.prototype.greet = function () {
+  return `${this.name} offers a greeting in ${this.language}.`;
+};
+
+
+// Stretch task: 
+
+  function Villian(villianAttrs){
+    Humanoid.call(this, villianAttrs);
+  };
+
+ Villian.prototype = Object.create(Humanoid.prototype);
+
+
+  Villian.prototype.attack = function () {
+    let villianAttack = Math.floor(Math.random() * 5) - this.healthPoints;
```

Comment:
This can be *const* if villianAttack is not reassigned elsewhere in the code block.